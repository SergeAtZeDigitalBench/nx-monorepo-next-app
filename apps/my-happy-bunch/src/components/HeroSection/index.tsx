import Image from 'next/image';

import { fetchHeroSection } from '../../lib/api';

type Props = {
  pathname: string;
};

const HeroSection = async ({ pathname }: Props) => {
  const [data, fetchError] = await fetchHeroSection(pathname);

  return (
    <section className=" bg-white px-2 py-8">
      <div className="min-h-[30vh] max-w-5xl mx-auto grid grid-cols-1 md:grid-cols-[2fr_1fr]">
        <div className=" h-full flex flex-col justify-center items-center">
          {data && (
            <>
              <h1 className="text-5xl font-bold my-4">{data.title}</h1>
              <p>{data.subTitle}</p>
            </>
          )}
          {fetchError && (
            <p className=" text-red-700 font-semibold text-center">
              {fetchError}
            </p>
          )}
        </div>
        <div className="hidden md:h-full md:flex md:flex-col justify-center items-center">
          {data && (
            <Image
              src={data.image.src}
              alt={data.image.alt}
              width={data.image.width}
              height={data.image.height}
              priority
            />
          )}
        </div>
      </div>
    </section>
  );
};

export const HeroSectionSkeleton = () => (
  <section className=" bg-white px-2 py-8">
    <div className="min-h-[30vh] max-w-5xl mx-auto grid grid-cols-1 gap-2 md:grid-cols-[2fr_1fr]  p-2 border-2 border-slate-300 rounded-lg">
      <div className=" h-full flex flex-col justify-start items-center">
        <div className=" bg-slate-200 h-[95px] rounded w-full my-2"> </div>
        <div className=" bg-slate-200  h-[50px] rounded w-full my-2"> </div>
      </div>
      <div className="hidden md:h-full md:flex md:flex-col justify-start items-center">
        <div className=" bg-slate-200 h-[200px] w-full rounded my-2"> </div>
      </div>
    </div>
  </section>
);

export default HeroSection;
