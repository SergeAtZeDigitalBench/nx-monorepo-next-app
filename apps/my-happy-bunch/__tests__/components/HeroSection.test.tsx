import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import HeroSection, {
  HeroSectionSkeleton,
} from '../../src/components/HeroSection';

describe('HeroSection', () => {
  describe('HeroSectionSkeleton', () => {
    it('should match the snapshot', () => {
      const { baseElement } = render(<HeroSectionSkeleton />);

      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('HeroSection', () => {
    it('should render the component', async () => {
      const { baseElement } = render(await HeroSection({ pathname: 'home' }));

      expect(baseElement).toBeDefined();
    });

    it('should render title subtitle and cover image', async () => {
      render(await HeroSection({ pathname: 'home' }));

      const title = screen.getByRole('heading', {
        name: /Welcome/i,
      });

      const subTitle = screen.getByText(/Nothing is more difficult/i);
      const image = screen.getByRole('img');

      expect(title).toBeDefined();
      expect(subTitle).toBeDefined();
      expect(image).toBeDefined();
    });
  });

  it('should handle error response case', async () => {
    render(await HeroSection({ pathname: 'somePageNotFound' }));

    const errorMessage = screen.getByText(/Hero section not found for the/i);
    expect(errorMessage).toBeDefined();
  });
});
