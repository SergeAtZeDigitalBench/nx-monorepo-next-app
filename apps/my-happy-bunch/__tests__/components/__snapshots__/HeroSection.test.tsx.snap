// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroSection > HeroSectionSkeleton > should match the snapshot 1`] = `
<body>
  <div>
    <section
      class=" bg-white px-2 py-8"
    >
      <div
        class="min-h-[30vh] max-w-5xl mx-auto grid grid-cols-1 gap-2 md:grid-cols-[2fr_1fr]  p-2 border-2 border-slate-300 rounded-lg"
      >
        <div
          class=" h-full flex flex-col justify-start items-center"
        >
          <div
            class=" bg-slate-200 h-[95px] rounded w-full my-2"
          >
             
          </div>
          <div
            class=" bg-slate-200  h-[50px] rounded w-full my-2"
          >
             
          </div>
        </div>
        <div
          class="hidden md:h-full md:flex md:flex-col justify-start items-center"
        >
          <div
            class=" bg-slate-200 h-[200px] w-full rounded my-2"
          >
             
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
